### PR TITLE
Inventorystat

### PR DESCRIPTION
**Pull request by EverettG**
Please Please Please I need it soo bad ooh ....ooh...OOOH MUH LIGHTFORCE

### CODE REVIEW

File: Assets/C#/Hittable.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ override public void OnInspectorGUI()
         }
     }
 
-    public enum DamageType { Neutral, Fire, Ice, Electric };
+	public enum DamageType { Neutral, Fire, Ice, Electric, Denim, Umbra };
```
Comment (Lonswaya) [inline Assets/C#/Hittable.cs:29]: > Denim

Yes

File: Assets/C#/Inventory.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
```
Comment (Lonswaya) [inline Assets/C#/Inventory.cs]: What was the reason for this again?

File: Assets/C#/InventoryController.cs [hunk 3]
```diff
@@ -5,26 +5,160 @@
...
+
+		if (i is Weapon) {
+			switch(((Weapon)i).weaponType){
+			case Weapon.WeaponCategory.Magic:
+				((Weapon)i).weaponSpeed += 1 * i.condition;
+				((Weapon)i).coolDown += 1 * i.condition;
+				((Weapon)i).range += 1 * i.condition;
+				((Weapon)i).damage += 3 * i.condition;
+					break;
+			case Weapon.WeaponCategory.Melee:
```
Comment (Lonswaya) [inline Assets/C#/InventoryController.cs:129]: We'll have to change these later in the weapons branch @AndrewGraber

Comment (Lonswaya) [reply Assets/C#/InventoryController.cs:129]: That is, if you renamed those

File: Assets/C#/InventoryController.cs [hunk 4]
```diff
@@ -5,26 +5,160 @@
...
+				break;
+			}
+
+		}
+
+
+	}
+
+	void useUpgradePotion(Pickup p){
+		switch (p.itemType) {
```
Comment (Lonswaya) [inline Assets/C#/InventoryController.cs:149]: Didn't we also want to use this for light force as well?

File: Assets/C#/StatsController.cs [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+	public void UpgradeMaxHealth(){
+		this.healthMax += HEALTH_INCREASE_AMOUNT;
+	}
+
+	public void UpgradeMaxMagic(){
+		this.magicMax += MAGIC_INCREASE_AMOUNT;
+	}
+
+	public void UpgradeMaxLightt(){
```
Comment (Lonswaya) [inline Assets/C#/StatsController.cs:86]: Lightt typo

File: Assets/C#/StatsController.cs [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+				return leftover;
+			}
+			magic += amount;
+			return 0;
+		}
+		return 0;
+	}
+
+	//Returns leftovers (if any) ((can be negative))
+	public float UpdateLightt(float amount) {
```
Comment (Lonswaya) [inline Assets/C#/StatsController.cs:141]: Lightt typo

File: Assets/C#/StatsController.cs [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+	/* Applies a change in damage from the vulnerability of the creature
+	 * to a certain DamageType and returns the damage after modification.
+	 * 
+	 * @param  damage - the amount of damage before modification
+	 * @param  type - The type of damage being dealt
+	 * 
+	 * @return The amount to modify damage being dealt by (damage is multiplied by this)
+	 */
+	public float ApplyDamageTypeHitMod(float damage, DamageType type) {
```
Comment (Lonswaya) [inline Assets/C#/StatsController.cs]: These should be private

File: Assets/C#/StatsController.cs [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+
+	/* Applies a reduction of damage from a hit based on the creature's armor, and returns
+	 * the damage after modification
+	 * 
+	 * @param  damage - the amount of damage before modification
+	 * @param  type - The type of damage being dealt
+	 * 
+	 * @return The damage to deal after modification
+	 */
+	public float ApplyArmorHitMod(float damage, DamageType type) {
```
Comment (Lonswaya) [inline Assets/C#/StatsController.cs]: These should be private

**Review by Lonswaya — CHANGES_REQUESTED** [review]

**Comment by Lonswaya** [comment]
Also, please pull from master and fix those conflicts

File: Assets/C#/Inventory.cs (hunk 2 above)
Comment (AndrewGraber) [reply Assets/C#/Inventory.cs]: Oh that was me. At one point I was using something from it, then it got deleted

**Review by Lonswaya — APPROVED** [review]

**Review by AndrewGraber — APPROVED** [review]
:)

**Pull request merged**